state_machine/applications/flight/lib/IGRF.py

```diff
@@ -0,0 +1,219 @@
+# flake8: noqa E741
+# Based on: https://github.com/RoboticExplorationLab/pycubed_circuitpython/blob/master/IGRF/igrf.py
+# Fifth order approximatin.
+
+try:
+    import ulab as np
+except ImportError:
+    import numpy as np
+import math
+
+def reset_array(input_array):
+    for i in range(len(input_array)):
+        input_array[i] = 0.0
+
+def unix_time_to_years_since_2020(unix_time):
+    twentytwenty = 1577854800  # 2022-01-01 00:00:00 time stamp
+    t = (unix_time - twentytwenty) / 31557600  # seconds in a Julian astronomical year
+    return t
+
+def igrf13_5(gh, date, latitude_degrees, elongitude_degrees, r_norm_km, cl, sl, p, q):
+
+    # reset the lists that are passed by reference
+    reset_array(cl)
+    reset_array(sl)
+    reset_array(p)
+    reset_array(q)
+
+    # colatitude
+    colat = 90 - latitude_degrees
+
+    # Declaration of variables
```

Comment:
Same thing here: maybe add inline comments for each of these explaining what they stand for, or (this is a future solution) add a comment that links users to the documentation that explains this